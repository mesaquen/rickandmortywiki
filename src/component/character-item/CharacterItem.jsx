import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Button from '../button/Button';

const Container = styled.div`
  display: flex;
  flex-direction: column;
  flex-shrink: 0;
  flex-grow: 0;
  padding: 4px;
`;

const Header = styled.header`
  display: flex
  max-width: 300px;
  background: red;
  position: relative;  
`;

const Title = styled.span`
  position: absolute;
  padding-left: 12px;
  bottom: 0;
  left: 0;
  font-size: 18px;
  font-weight: bold;
  color: white;
  background: rgba(0, 0, 0, 0.3);
  width: 100%;
`;

const Image = styled.img`
  display: flex;
`;

const Label = styled.span`
  font-size: 14px;
`;

const ItemButton = styled(Button)`
  margin-top: 8px;
`;

export default class CharacterItem extends PureComponent {
  static propTypes = {
    item: PropTypes.object,
  };
  render() {
    const { item } = this.props;
    return (
      <Container>
        <Header>
          <Image src={item.image} />
          <Title>{item.name}</Title>
        </Header>

        <Label>Species: {item.species}</Label>
        <Label>Gender: {item.gender}</Label>
        <Label>Origin: {item.origin.name}</Label>
        <ItemButton>Details</ItemButton>
      </Container>
    );
  }
}
