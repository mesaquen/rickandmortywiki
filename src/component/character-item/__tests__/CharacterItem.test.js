import React from 'react';
import Renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import CharacterItem from '../CharacterItem';
/**
 * @author Mesaque Francisco <mesaquenf@gmail.com>
 * @since 2019-11-16
 */

describe('CharacterItem', () => {
  let component = Renderer.create(<CharacterItem loading />);
  it('should have same snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  describe('when passing valid values', () => {
    const callback = jest.fn();
    const item = {
      name: 'name',
      species: 'species',
      gender: 'gender',
      origin: { name: 'origin' },
    };
    beforeEach(() => {
      component = Renderer.create(
        <CharacterItem item={item} onClickDetails={callback} />
      );
    });

    it('should have same snapshot', () => {
      expect(component.toJSON()).toMatchSnapshot();
    });

    it('should call onCllickDetails with same value', () => {
      const { getByTestId } = render(
        <CharacterItem item={item} onClickDetails={callback} />
      );
      const element = getByTestId('button');
      fireEvent.click(element);

      expect(callback.mock.calls).toHaveLength(1);
      expect(callback.mock.calls[0][0]).toEqual(item);
    });
  });
});
