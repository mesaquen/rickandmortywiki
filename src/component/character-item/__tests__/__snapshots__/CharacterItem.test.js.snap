// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterItem should have same snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-egiyK hhJPHS"
  >
    <header
      class="sc-bqiRlB bLBhLX"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="width: 300px; height: 300px; --highlight-color: #AAA;"
        >
          ‌
        </span>
        <br />
      </span>
    </header>
    <div
      class="sc-fFeiMQ bIjmor"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="--highlight-color: #AAA;"
        >
          ‌
        </span>
        <br />
        <span
          class="react-loading-skeleton"
          style="--highlight-color: #AAA;"
        >
          ‌
        </span>
        <br />
        <span
          class="react-loading-skeleton"
          style="--highlight-color: #AAA;"
        >
          ‌
        </span>
        <br />
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CharacterItem when passing valid values should have same snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-egiyK hhJPHS"
  >
    <header
      class="sc-bqiRlB bLBhLX"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="width: 300px; height: 300px; --highlight-color: #AAA;"
        >
          ‌
        </span>
        <br />
      </span>
      <img
        class="sc-gsDKAQ hrqDZy"
        height="300"
        width="300"
      />
      <span
        class="sc-dkPtRN jkAsmV sc-ksdxgE idArhN"
        color="#444"
      >
        name
      </span>
    </header>
    <div
      class="sc-fFeiMQ bIjmor"
    >
      <div
        class="sc-hKwDye dodXuz"
      >
        <span
          class="sc-dkPtRN iBkRmY"
          color="#999"
        >
          Species
        </span>
        <span
          class="sc-dkPtRN uImFO sc-eCImPb gVHibD"
          color="teal"
        >
          species
        </span>
      </div>
      <div
        class="sc-hKwDye dodXuz"
      >
        <span
          class="sc-dkPtRN iBkRmY"
          color="#999"
        >
          Gender
        </span>
        <span
          class="sc-dkPtRN uImFO sc-eCImPb gVHibD"
          color="teal"
        >
          gender
        </span>
      </div>
      <div
        class="sc-hKwDye dodXuz"
      >
        <span
          class="sc-dkPtRN iBkRmY"
          color="#999"
        >
          Origin
        </span>
        <span
          class="sc-dkPtRN uImFO sc-eCImPb gVHibD"
          color="teal"
        >
          origin
        </span>
      </div>
    </div>
    <button
      class="sc-bdvvtL lbBTIY sc-fotOHu llQTdp"
      color="teal"
      data-testid="button"
    >
      Details
    </button>
  </div>
</DocumentFragment>
`;
